tests/linalg_test.py

```diff
@@ -534,6 +534,70 @@ def testQrBatching(self, shape, dtype, rng_factory):
     qs, rs = vmap(jsp.linalg.qr)(args)
     self.assertTrue(onp.all(onp.linalg.norm(args - onp.matmul(qs, rs)) < 1e-3))
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name":
+       "_shape={}".format(jtu.format_shape_dtype_string(shape, dtype)),
+       "shape": shape, "dtype": dtype, "rng_factory": rng_factory}
+      for shape in [(1, 1), (4, 4), (2, 3, 5), (5, 5, 5), (20, 20), (5, 10)]
+      for dtype in float_types + complex_types
+      for rng_factory in [jtu.rand_default]))
+  def testCond(self, shape, dtype, rng_factory):
+    _skip_if_unsupported_type(dtype)
+    rng = rng_factory()
+
+    def gen_mat():
+      arr = rng(shape, dtype)
+      nan_arr = np.full_like(arr, np.nan)
+      mask = onp.random.choice([True, False], size=shape, p=[1./2, 1./2])
+      res = np.where(mask, nan_arr, arr)
+      return res
+
+    def args_gen(p):
+      def _args_gen():
+        return [gen_mat(), p]
+      return _args_gen
+
+    norm_types = [-np.inf, 1, -1, 2, -2, 'fro']
+    for pnorm in norm_types:
```

Comment:
Please move this loop into the `named_parameters`. Avoiding control flow in tests (when possible) makes test failures easier to debug.

Reply:
Ok, have done this.